inc/class-tachyon.php

```diff
@@ -584,14 +584,26 @@ public function filter_image_downsize( $image, $attachment_id, $size ) {
 				 */
 				$tachyon_args = apply_filters( 'tachyon_image_downsize_string', $tachyon_args, compact( 'image_args', 'image_url', 'attachment_id', 'size', 'transform' ) );
 
-				// Generate Tachyon URL.
+				// Calculate the correct target width & height.
 				// We want the width / height params to match the dimensions of the image,
 				// not the resize dimensions. The Resize dimensions might be "Max width" /
 				// "Max-height" dimensions, rather than absolute image size.
+				$target_width  = $image_args['width'];
+				$target_height = $image_args['height'];
+				if ( ! $image_args['crop'] ) {
+					list( $target_width, $target_height ) = wp_constrain_dimensions(
```

Comment:
Hmm `$image_meta` is meant to be the meta for that specific size (see https://github.com/humanmade/tachyon-plugin/pull/42/files#diff-6e50cef3856fe5b55cd1e3112bbdab09R495), not the size of the original image. Are you not seeing that be the case?

The 'true' size of the intermediate image is stored in the attachment metadata, and that's the data I was intending to use.

Or, is this trying to work in cases where there is no size data in the attachment metadata?